src/Microsoft.Extensions.DependencyModel/Resolution/PackageCacheCompilationAssemblyResolver.cs

```diff
@@ -42,37 +41,19 @@ public bool TryResolveAssemblyPaths(CompilationLibrary library, List<string> ass
 
             if (!string.IsNullOrEmpty(_packageCacheDirectory))
             {
-                var hashSplitterPos = library.Hash.IndexOf('-');
```

Comment:
The full fix for the issue of the host and DependencyModel being out of sync should be addressed by https://github.com/dotnet/core-setup/issues/624#issuecomment-288245699